### PR TITLE
CRN-682 Search across users and external authors

### PR DESCRIPTION
**Pull request by niezgoda**
## Requirements:
- on the Share an Output form, in the author selector, show items which can be either users or external authors 
- display  external authors differently from users:
- external authors 's names should be followed by “(Non CRN)”
- be able to select from at least 3 existing external authors 
- search what is typed on the author selector against these fields only:
  - display name of  external authors 
  - first name of  users 
  - last name of users 
- when user clicks "Share", all users and external authors   selected must be added to the "Authors" field on the CMS
- the order of users and external authors must be remembered
- when user clicks "Share", this field must become disabled (as the page is saving)
- testing strategy: on the FE, it will be possible to find and select those  external authors 

## OOS:
- keep external authors in sync (manually or automatically)

Previous PR: https://github.com/yldio/asap-hub/pull/1423

### CODE REVIEW

**Review by peterstarling — APPROVED** [review]

File: apps/crn-frontend/src/__fixtures__/algolia.ts [hunk 1]
```diff
@@ -21,11 +22,14 @@ export const createAlgoliaResponse = <EntityType extends keyof EntityResponses>(
...
-    objectID: `${i}`,
-    __meta: { type },
-  })),
+  hits: data.map(
+    (item, i) =>
+      ({
+        ...item,
+        objectID: `${i}`,
+        __meta: { type },
+      } as EntityHit<EntityType>),
```
[the remark this answers is not in the document]
Comment (niezgoda) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:31]: We have to. Otherwise TS is screaming on us.

[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:31]: to be fair i noticed this and was going to suggest setting the return type for the map as well but these are just fixtures so i dont mind the cast as much anyway

File: apps/crn-frontend/src/network/users/api.ts [hunk 2]
```diff
@@ -48,6 +50,24 @@ export const getUsers = async (
   return { items: result.hits, total: result.nbHits };
 };
 
+export const getUsersOrExternalAuthors = async (
```
Comment (kmaid) [inline apps/crn-frontend/src/network/users/api.ts:53]: Shouldn't this be getUsersAndExternalAuthours?

File: apps/crn-frontend/src/__fixtures__/algolia.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (niezgoda) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:31]: @peterstarling that is not going to work :(

[the remark this answers is not in the document]
Comment (niezgoda) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:31]: @kmaid - what solution are you suggesting?
We have trick there to enforce that data and __meta.type is same type.
Eg. ...items is RO and __meta.type must be "research-output"

File: apps/crn-frontend/src/__fixtures__/algolia.ts [hunk 3]
```diff
@@ -21,11 +22,14 @@ export const createAlgoliaResponse = <EntityType extends keyof EntityResponses>(
   renderingContent: {},
   processingTimeMS: 1,
   ...overrides,
-  hits: data.map((item, i) => ({
-    ...item,
-    objectID: `${i}`,
-    __meta: { type },
-  })),
+  hits: data.map(
+    (item, i) =>
```
Comment (kmaid) [inline apps/crn-frontend/src/__fixtures__/algolia.ts:26]: > (item, i): EntityHit<EntityType> =>

Comment (peterstarling) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:26]: yeah that would only work if the input types were dont differently - ie the `data` and `type` were actually tied together. Technically at the moment you can give it a type that does not correspond to the data which is why typescript moans. Like I said its just a fixture so I wouldn't get too hung up on it and simply leave the cast

Comment (niezgoda) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:26]: That's not going to work

Comment (niezgoda) [reply apps/crn-frontend/src/__fixtures__/algolia.ts:26]: @kmaid If you have better solution on your mind please push it into this branch/PR :)

**Review by peterstarling — APPROVED** [review]

**Review by kmaid — APPROVED** [review]

**Pull request merged**